services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/TopicManager.java

```diff
@@ -15,9 +15,72 @@
 */
 package com.futurewei.alcor.dataplane.client.pulsar;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+@Configuration
+@ConditionalOnProperty(prefix = "mq", name = "type", havingValue = "pulsar")
 public class TopicManager {
+    private static final Logger LOG = LoggerFactory.getLogger(TopicManager.class);
+
+    @Autowired
+    private PulsarConfiguration configuration;
+
+    private Map<String, String> hostIpToGroupTopic;
```

Comment:
Let us plan to build these mapping from node metadata manager and store it in the local cache under development.